finat/fiat_elements.py

```diff
@@ -205,16 +215,21 @@ def point_evaluation_ciarlet(fiat_element, order, refcoords, entity):
             D = form_matrix_product(poly_set.get_dmats(), alpha)
             table = np.dot(poly_set.get_coeffs(), np.transpose(D))
             assert table.shape[-1] == m
-            beta = tuple(gem.Index() for s in table.shape[:-1])
-            k = gem.Index()
-            result[alpha] = gem.ComponentTensor(
-                gem.IndexSum(
-                    gem.Product(gem.Indexed(gem.Literal(table), beta + (k,)),
-                                gem.Indexed(base_values, (k,))),
-                    (k,)
-                ),
-                beta
-            )
+            zerocols = np.isclose(abs(table).max(axis=tuple(range(table.ndim - 1))), 0.0)
```

Comment:
Yes, because `max(-1.2, 0.01) == 0.01` so it would otherwise fail to check closeness to zero for negative numbers.